src/ImageZoom.js

```diff
@@ -85,9 +85,12 @@ export default class ImageZoom extends Component {
       this.isClosing = true
     }
 
+    const currentSource = this.props.image.src
+    const nextSource = nextProps.image.src
+
     // If the consumer wants to change the image's src, then so be it.
-    if (this.props.image.src !== nextProps.image.src) {
-      this.setState({ image: nextProps.image })
+    if (currentSource !== nextSource) {
+      this.setState({ source: nextSource })
```

Comment:
I think it'd be wise to keep using `src` instead of source. This would change this bit of code like so:
```js
const { src } = this.props.image
const { src: nextSrc } = nextProps.image

if (src !== nextSrc) {
  this.setState({ src: nextSrc })
}
```
In most cases, I prefer words to be spelled out; however, for this one, I like `src` best because it's also the property name

Reply:
Sure thing. I went back and forth on this one. I went the direction I did so that it was distinguished from the element prop, but don't have super strong feelings about it either way.